backend/functions/lib/use-cases/case.assignment.ts

```diff
@@ -4,72 +4,81 @@ import { CaseAttorneyAssignment } from '../adapters/types/case.attorney.assignme
+    caseId: string,
+    newAssignments: string[],
+    role: string,
   ): Promise<AttorneyAssignmentResponseInterface> {
-    const isValid = await this.isCreateValid(listOfAssignments);
-    if (!isValid) {
-      throw new AssignmentError(MODULE_NAME, { message: EXISTING_ASSIGNMENT_FOUND });
-    } else {
-      const listOfAssignmentIdsCreated: string[] = [];
-      for (const assignment of listOfAssignments) {
-        const assignmentId = await this.createAssignment(assignment);
-        if (!listOfAssignmentIdsCreated.includes(assignmentId))
-          listOfAssignmentIdsCreated.push(assignmentId);
-      }
-
-      log.info(
-        applicationContext,
-        MODULE_NAME,
-        `Created ${listOfAssignmentIdsCreated.length} assignments for case number ${listOfAssignments[0].caseId}.`,
-        listOfAssignmentIdsCreated,
-      );
-      return {
-        success: true,
-        message: 'Trial attorney assignments created.',
-        count: listOfAssignmentIdsCreated.length,
-        body: listOfAssignmentIdsCreated,
+    const listOfAssignments: CaseAttorneyAssignment[] = [];
+    const attorneys = [...new Set(newAssignments)];
+    const currentDate = new Date().toISOString();
+    attorneys.forEach((attorney) => {
+      const assignment: CaseAttorneyAssignment = {
+        caseId: caseId,
+        name: attorney,
+        role: CaseAssignmentRole[role],
+        assignedOn: currentDate,
       };
+      listOfAssignments.push(assignment);
+    });
+    const listOfAssignmentIdsCreated: string[] = [];
+
+    // Unassign an existing attorney that does not appear in the new assignment list.
+    const existingAssignments = await this.assignmentRepository.findAssignmentsByCaseId(caseId);
+    for (const existingAssignment of existingAssignments) {
+      const stillAssigned = listOfAssignments.find((newAssignment) => {
+        return (
+          newAssignment.name === existingAssignment.name &&
+          newAssignment.role === existingAssignment.role
+        );
+      });
+      if (!stillAssigned) {
+        await this.assignmentRepository.updateAssignment({
+          ...existingAssignment,
+          unassignedOn: new Date().toISOString(),
+        });
+      }
     }
-  }
 
-  public async isCreateValid(newAssignments: CaseAttorneyAssignment[]): Promise<boolean> {
-    const caseId = newAssignments[0].caseId;
-    try {
-      const existingAssignments = await this.assignmentRepository.findAssignmentsByCaseId(caseId);
-      return existingAssignments.length === 0;
-    } catch (e) {
-      throw new AssignmentError(MODULE_NAME, {
-        message:
-          'Unable to determine whether assignments already exist. Please try again later. If the problem persists, please contact USTP support.',
-        originalError: e,
-        status: 500,
+    log.info(applicationContext, MODULE_NAME, 'New assignments:', newAssignments);
+
+    // Add any attorney from the new assignment list to the case that is not already assigned.
```

Comment:
Do we feel we need this comment? If so, maybe we should just extract the relevant code into a named function so that we don't need a comment.